bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/inbox/InboxFilterCriteria.java

```diff
@@ -26,61 +27,137 @@
     private final String bindingId;
     private final ThingTypeUID thingTypeUID;
     private final ThingUID thingUID;
+    private final String representationValue;
```

Comment:
If we leave it unchanged, we can deprecate the whole InboxFilterCriterla class right now, so nothing would be left in an 1.0 release.

>  And how to introduce the predicates then?

Similar to the Registry: The PersistentInbox should get a new `stream()` method besides the current `get` and `getAll`. But you can also ask @ViToni for help as he designed this all :-)